blueocean-commons/src/main/java/io/jenkins/blueocean/commons/analytics/Analytics.java

```diff
@@ -0,0 +1,70 @@
+package io.jenkins.blueocean.commons.analytics;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Maps;
+import com.google.common.hash.Hashing;
+import hudson.model.UsageStatistics;
+import hudson.model.User;
+import io.jenkins.blueocean.commons.ServiceException;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Restricted(NoExternalUse.class)
+public abstract class Analytics {
```

Comment:
yeah but sounds like right thing to do without much cost. but I agree no immediate need of seeing it as extension point that is fairly easily replaceable, might defeat the purpose. I guess when this happens it can be big deal anyways. I will leave it up to you.